common/lib/dependabot/clients/bitbucket.rb

```diff
@@ -52,6 +68,92 @@ def fetch_file_contents(repo, commit, path)
         response.body
       end
 
+      def commits(repo, branch_name = nil)
+        commits_path = "#{repo}/commits/" + branch_name.to_s
+        commits_path += "?pagelen=100"
+
+        response = get(base_url + commits_path)
+
+        JSON.parse(response.body).fetch("values")
+      end
+
+      def branch(repo, branch_name)
+        branch_path = "#{repo}/refs/branches/#{branch_name}"
+        response = get(base_url + branch_path)
+
+        JSON.parse(response.body)
+      end
+
+      def pull_requests(repo, source_branch, target_branch)
+        pr_path = "#{repo}/pullrequests"
+        pr_path += "?status=OPEN&status=MERGED&status=DECLINED&status=SUPERSEDED"
+        next_page = base_url + pr_path
+        pull_requests = []
+        loop do
+          response = get(next_page)
+          result = JSON.parse(response.body)
+          pull_requests.concat result.fetch("values")
+          break unless result.key?("next")
+
+          next_page = result.fetch("next")
+        end
+
+        pull_requests unless source_branch && target_branch
+
+        pull_requests.
+          select do |pr|
+            pr_source_branch = pr.fetch("source").fetch("branch").fetch("name")
+            pr_target_branch = pr.fetch("destination").fetch("branch").fetch("name")
+            pr_source_branch == source_branch && pr_target_branch == target_branch
+          end
+      end
+
+      # rubocop:disable Metrics/ParameterLists
+      def create_commit(repo, branch_name, base_commit, commit_message, files,
+                        author_details)
+        parameters = {
+          message: commit_message, # TODO: Format markup in commit message
+          author: "#{author_details.fetch(:name)} <#{author_details.fetch(:email)}>",
+          parents: base_commit,
+          branch: branch_name
+        }
+
+        files.each do |file|
+          absolute_path = "/" + file.name unless file.name.start_with?("/")
```

Comment:
Concern: what value should `absolute_path` take if `file.name.start_with?("/") == true`?
It might be `nil` currently, which would mean multiple absolute paths conflict.

Reply:
Ah, that was a mistake; I've corrected that.